index.js

```diff
@@ -177,27 +177,55 @@ Funnel.prototype.build = function() {
   if (this.shouldLinkRoots()) {
     linkedRoots = true;
 
-    var inputPathExist = fs.existsSync(inputPath);
-
-    if (this._lastInputPath ? !inputPathExist : inputPathExist) {
-      rimraf.sync(this.outputPath);
-
-      if (inputPathExist) {
+    /**
+     * We want to link the roots of these directories, but there are a few
+     * edge cases we must account for.
+     *
+     * 1. It's possible that the original input doesn't actually exist.
+     * 2. It's possible that the output symlink has been broken.
+     * 3. We need slightly different behavior on rebuilds.
+     *
+     * Behavior has been modified to always having an `else` clause so that
+     * the code is forced to account for all scenarios. Not accounting for
+     * all scenarios made it possible for initial builds to succeed without
+     * specifying `this.allowEmpty`.
+     */
+
+    var inputPathExists = fs.existsSync(inputPath);
+    var outputPathExists = fs.existsSync(this.outputPath);
```

Comment:
Hrm. Tests still pass using `exists-sync`. Not sure if that's a gap in our coverage or... ?